### PR TITLE
[draft] Add load balancer server type

### PR DESCRIPTION
**Pull request by austinkregel**
This PR aims to add a new server type for load balancing with nginx. Inspired in [part by this discussion thread](https://github.com/vitodeploy/vito/discussions/241), and my personal usage of load balancing servers 😄 

**Goal**: Deploy a Load Balancing server from the UI

**Acceptance Criteria**:
 - [x] (done) Have "load balancing" be a new server type, that configures only an nginx server
image

 - [ ] (in progress) Have all sites under a "load balancing" server change their "application" page to reflect the load-balancer configuration.
 - [ ] (todo) Only allow load balancing to privately networked servers. (I personally use [Zerotier](https://www.zerotier.com/), when I'm not using DigitalOcean)
 - [ ] (todo) Allow configuring of the [load balancing method](https://docs.nginx.com/nginx/admin-guide/load-balancer/http-load-balancer/#choosing-a-load-balancing-method) (round robin, ip_hash, hash, last_conn)
 - [ ] (todo) Allow configuring the host and port of the servers we're balancing to.
 - [ ] (todo) Allow setting server as "backup" or as "down"

Optional concepts to explore:
 - Having the load balancing server auto-provision the site on servers you connect to. This would require defining the project repo, deployment script, php version, project directory, and any extra fields based on the site type when creating the site on the load balancing server.
 - 

Old Laravel Forge load balancing config page to convey basic page layout. Exact style, format, or design will vary. 
image

### CODE REVIEW

**Comment by rtraselbd** [comment]
Go ahead, waiting for it.

File: app/Http/Controllers/SiteController.php [hunk 1]
```diff
@@ -61,6 +62,13 @@ public function show(Server $server, Site $site, Request $request): View|Redirec
             return redirect()->route('servers.sites.installing', [$server, $site]);
         }
 
+        if ($server->type === ServerType::LOAD_BALANCER) {
```
Comment (saeedvaziry) [inline app/Http/Controllers/SiteController.php:65]: I wouldn't use Sites to handle the load balancer configs.

I would go with a new table and controller and model and ... specifically designed for this purpose.

**Comment by austinkregel** [comment]
I've fallen into a huge rabbit hole learning about Bison and language parsing 😵 I found an [article about using bison and PHP to parse the Nginx config](https://dev.to/mrsuh/nginx-parser-with-php-and-bison-1k5) into [AST](https://en.wikipedia.org/wiki/Abstract_syntax_tree) and have been able to extend the project and I might be able to fully support all nginx directives :eyes: 

With one of the goal being more advanced [nginx config parsing](https://github.com/austinkregel/php-nginx-ast-parser) and eventually editing so projects like Vito can work with any existing config. 

With that said, I've lost ~~about 2 weeks in this hole 🙈~~ a little over half a week but it feels like longer 😂 I need a bit of a break to re-calibrate, and reset to tackle this specifically and not nginx parsing as a whole.

**Comment by saeedvaziry** [comment]
@austinkregel first of all thanks for this contribution, I am sure it will be a big update.

second, I am sure there is easier workarounds and personally don't think parsing nginx files is really needed.

Feel free to reach me on Discord to have a chat about it.

**Comment by saeedvaziry** [comment]
@austinkregel shall we close this PR?

**Pull request closed without merging**